wolfcrypt/src/ascon.c

```diff
@@ -0,0 +1,528 @@
+/* ascon.c
+ *
+ * Copyright (C) 2006-2025 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#ifdef HAVE_ASCON
+
+#include <wolfssl/wolfcrypt/ascon.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+/*
+ * Implementation of the ASCON AEAD and HASH algorithms. Based on the NIST
+ * Initial Public Draft "NIST SP 800-232 ipd" and reference implementation found
+ * at https://github.com/ascon/ascon-c.
+ */
+
+/*
+ * TODO
+ * - Add support for big-endian systems
+ * - Add support for 32-bit and smaller systems */
+
+#define MAX_ROUNDS 12
+
+#ifndef WOLFSSL_ASCON_UNROLL
+
+/* Table 4 */
+static const byte round_constants[MAX_ROUNDS] = {
+    0xf0, 0xe1, 0xd2, 0xc3, 0xb4, 0xa5, 0x96, 0x87, 0x78, 0x69, 0x5a, 0x4b
+};
+
+static byte start_index(byte rounds)
+{
+    switch (rounds) {
+        case 6:
```

Comment:
I can't see where rounds will be 6.

Reply:
Removing as its currently not used in either algo.